tox.ini

```diff
@@ -11,6 +11,12 @@ envlist =
     flake8-docstrings
     pre-commit
 
+[gh-actions]
+python =
+    3.8: py38
+    3.9: py39
+    3.10: py310
+    3.11: py311, cercis, check-self, flake8-basic, flake8-misc, flake8-docstrings, pre-commit
```

Comment:
Shall we do `cercis, check-self, flake8-basic, flake8-misc, flake8-docstrings, pre-commit` for py38, py39, and py310 as well?

Right now, only in Py311 does the pipeline do `cercis, check-self, flake8-basic, flake8-misc, flake8-docstrings, pre-commit` checks.

Reply:
That would be testing that those tools work the same across python versions, but what those tools should do is check the code of `pydoclint`.

Personally, I would move them all into `pre-commit` since that would prevent commits like `Fixing flake8 issues`.

I also like to run `pre-commit` as a separate CI job and make actual tests depend on it (won't run on fail) since QA tool violations require additional commits and CI runs anyway.

In action, the flow looks something like this where the lines indicate dependencies
![image](https://github.com/jsh9/pydoclint/assets/9513634/14580b5f-c649-4d76-9ff4-45014171ab39)

Reply:
There are things in the pydoclint code that are dependent on Python versions (such as [here](https://github.com/jsh9/pydoclint/blob/f938a5afd47205179327ad9bbb48117a3f3be849/pydoclint/utils/annotation.py#L20-L38)).  So I think it's safer if we just run all the tox checks in all Python versions, even if some of them may be redundant.